controllers/localVolumeCleanUp.go

```diff
@@ -1,76 +1,119 @@
-// 			return false
-// 		}
-// 	}
-// 	fmt.Println("PV Deleted.....")
-// 	return true
-// }
+func (cr *CleanUpOperatorReconciler) localVolumeCleanUp(ctx context.Context, namespace string) error {
+	localVolumeStatus := true
+	localDisk := &localv1.LocalVolume{}
+	err := cr.Get(ctx, types.NamespacedName{Name: "local-disk", Namespace: namespace}, localDisk)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Println("LocalVolume 'local-disk' not found")
+			localVolumeStatus = false
+		} else {
+			fmt.Println(err, "Error in getting LocalVolume 'local-disk'")
+			return err
+		}
+	}
 
-// // deleteMountedPath deletes mounted path from each node
-// func (r *CleanUpWatcher) deleteMountedPath(ctx context.Context) bool {
-// 	// Remove Mounted Path
-// 	nodesList := &corev1.NodeList{}
-// 	err := r.List(ctx, nodesList)
-// 	if err != nil {
-// 		fmt.Print("Error in Getting Nodes List")
-// 		return false
-// 	}
+	if localVolumeStatus {
+		localDisk.SetFinalizers([]string{})
+		if err := cr.Update(ctx, localDisk); err != nil {
+			fmt.Println(err, "Error is removing finalizers from Local-Volume 'local-disk'")
+			return err
+		}
+	}
 
-// 	for _, node := range nodesList.Items {
-// 		command := "oc debug node/" + node.Name + " -- chroot /host rm -rf /mnt"
-// 		_, out, _ := ExecuteCommand(command)
-// 		fmt.Println(out)
-// 	}
-// 	fmt.Println("Mounted Paths Removed....")
-// 	return true
-// }
+	// Find PVs
+	pvList := &corev1.PersistentVolumeList{}
+	pvfound := true
+	err = cr.List(ctx, pvList)
+	if err != nil {
+		fmt.Println(err, "Error in getting PVs")
+		pvfound = false
+	}
 
-// // localVolumeNSCleanUp performs cleanUp when namespace is in terminating state
-// func (r *CleanUpWatcher) localVolumeNSCleanUp(ctx context.Context, namespace string, resources []map[string]string, flag bool) bool {
-// 	patchFinalizer("localvolumes.local.storage.openshift.io", "local-disk", namespace)
-// 	if r.fetchRemovePV(ctx) && r.deleteMountedPath(ctx) {
-// 		if flag {
-// 			for index := range resources {
-// 				resourceType := resources[index]["Type"]
-// 				resourceName := resources[index]["Name"]
-// 				resourceNamespace := resources[index]["Namespace"]
-// 				if resourceType == "deployment" {
-// 					patchFinalizer(resourceType, resourceName, resourceNamespace)
-// 				}
-// 			}
-// 		}
-// 		return true
-// 	}
-// 	return false
-// }
+	// PV Deletion
+	if pvfound {
+		for _, pv := range pvList.Items {
+			if strings.HasPrefix(pv.Name, "local-pv-") {
+				fmt.Println("PV status- ", pv.Status.Phase)
+				// if pv.Status.Phase == "Available" {
+				err = cr.Delete(ctx, &pv)
+				if err != nil {
+					fmt.Print("Error in Deleting PV ", pv.Name)
+					return err
+				}
+				// }
+			}
+		}
+		fmt.Println("PV(s) Deleted.....")
+	}
+
+	// Remove Mounted Path
+	nodesList := &corev1.NodeList{}
+	nodesfound := true
+	err = cr.List(ctx, nodesList)
+	if err != nil {
+		fmt.Println(err, "Error in getting Nodes List")
+		nodesfound = true
```

Comment:
Similarly here.
I think we need to return err if there is any error while getting PV list.
I asked just not to remove IsNotFound error